test/views/honeysql/util_test.clj

```diff
@@ -65,4 +65,56 @@
   (testing "will return table mentioned in select clause"
     (let [query {:select [[{:select [1] :from [:foo] :limit 1} :col]] :from [:bar]}]
       (is (= #{:foo :bar}
+             (util/query-tables query)))))
+
+  (testing "multiple from,join"
+    (let [query {:select [[{:select [:%count.aa.a]
+                            :from [[:AA :aa]
+                                   [:BB :bb]
+                                   [:CC :cc]]} :ca]
+                          [{:select [:%count.dd.a]
+                            :from {:DD :dd}} :cd]]
+                 :from {:EE :ee}
+                 :join [[:FF :ff] [:= :ff.a :ee.a]
+                        [:GG :gg] [:= :gg.a :ee.a]]}]
+      (is (= #{:AA :BB :CC :DD :EE :FF :GG}
+             (util/query-tables query)))))
+
+  (testing "DEV-9832"
```

Comment:
It's fine to leave references to JIRA tickets but since this is an external repo, it would benefit from a short description of the problem. Something like "includes tables in COUNT ... FILTER".